getconf/base.py

```diff
@@ -39,167 +34,27 @@ def __hash__(self):
             default = tuple(self.default)
         else:
             default = self.default
-        return hash((self.section, self.entry, self.envvar, self.doc, default, self.type_hint))
-
-    def as_ini_entry(self):
-        """Format as a commented entry in INI format (section is omitted)
-
-        ; type=str - The provided documentation
-        ;entry = default_value
-        """
-        # Remove any new line in self.doc
-        one_line_doc = ' '.join(self.doc.split())
-        doc_part = (' - %s' % one_line_doc) if one_line_doc else ''
-
-        if self.type_hint == 'list':
-            default = ', '.join(self.default)
-        elif self.type_hint == 'bool':
-            default = 'on' if self.default else 'off'
-        else:
-            default = str(self.default)
-        default_str = ' %s' % default if default else ''
-        return '; {envvar} - type={type_hint}{doc_part}\n;{entry} ={default_str}'.format(
-            envvar=self.envvar,
-            type_hint=self.type_hint,
-            doc_part=doc_part,
-            entry=self.entry,
-            default_str=default_str,
-        )
-
-
-class NotFound(KeyError):
-    """Raised when a key is not found in a configuration source."""
-
-
-class ConfigGetter(object):
-    """A simple wrapper around ConfigParser + os.environ.
-
-    Designed for use in a settings.py file.
+        return hash((self.key, self.doc, default, self.type_hint))
 
-    Usage:
-        >>> config = ConfigGetter('blusers', ['/etc/blusers/settings.ini'])
-        >>> x = config.get('psql.server', 'localhost:5432')
-        'localhost:5432'
-
-    With the above ``ConfigGetter``:
-    - Calls to get('psql.server', 'foo') will look at:
-        - Environment key BLUSERS_PSQL_SERVER
-        - Key 'server' from section 'pqsl' of config file given in env
-          BLUSERS_CONFIG, if provided
-        - Key 'server' from section 'psql' of config file
-          ``/etc/blusers/settings.ini``
-        - Key 'psql' in key 'server' of default config dict
-        - The provided default
 
-    - Calls to get('secret_key') will look at:
-        - Environment key BLUSERS_SECRET_KEY
-        - Key 'secret_key' from section 'DEFAULT' of config file given in env
-          BLUSERS_CONFIG, if provided
-        - Key 'secret_key' from section 'DEFAULT' of config file
-          ``/etc/blusers/settings.ini``
-        - Key 'secret_key' of default config dict
-        - The empty string
-    """
+class CustomConfigGetter(object):
 
-    def __init__(self, namespace, config_files=(), defaults=None):
-        self.parser = compat.get_no_interpolation_config_parser()
+    def __init__(self, *finders_):
```

Comment:
Using `config_finders` to name the args could allow you to not use this ugly ending underscore.